x/authenticator/authenticator/any_of.go

```diff
@@ -124,10 +120,10 @@ func (aoa AnyOfAuthenticator) Track(ctx sdk.Context, account sdk.AccAddress, msg
 }
 
 func (aoa AnyOfAuthenticator) ConfirmExecution(ctx sdk.Context, account sdk.AccAddress, msg sdk.Msg, authenticationData iface.AuthenticatorData) iface.ConfirmationResult {
-	// Call ConfirmExecution on executed sub-authenticators
-	for _, auth := range aoa.executedAuths {
-		if confirmation := auth.ConfirmExecution(ctx, nil, msg, authenticationData); confirmation.IsBlock() {
-			return confirmation
+	for _, auth := range aoa.SubAuthenticators {
+		result := auth.ConfirmExecution(ctx, account, msg, authenticationData)
+		if result.IsBlock() {
```

Comment:
open question: Here we're doing the same on Any and All. That is: all sub authenticators must confirm.

Semantically it makes sense for "any" to only need one subAuthenticator to confirm. Is that what we want?

Any(
  spend_limit(100, month)
  spend_limit(200 day)
)

would ignore the 100 spend limit in this case (since there is no case where it would fail and the 200 wouldn't). 

Do you think it makes sense to always treat confirm as an "all" (which is pretty much what we do at the top level) or should it be and "any" here too?

Reply:
I think it makes sense to treat confirm as an all.

e.g
```
Any(
spend_limit(100, month)
spend_limit(100, month)
spend_limit(100, month)
)
```
===
```
Any(
spend_limit(100, month)
)
```

Or a case like this would make more sense:
```
Any(
signature_verification()
spend_limit(100, month)
)
```

even if the signature passes we update the spend limit.

wdyt?

Reply:
from our discussion on slack. Let's make this work as it would be expected on an Any: if at least one of the authenticators confirm, return confirm